app/controllers/api/v2/ansible_playbooks_controller.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    # API controller for Ansible Roles
+    class AnsiblePlaybooksController < ::Api::V2::BaseController
+      include ::Api::Version2
+
+      resource_description do
+        api_version 'v2'
+        api_base_url '/ansible/api'
+      end
+
+      before_action :find_proxy, :only => [:fetch, :sync]
+      before_action :create_importer, :only => [:fetch, :sync]
+
+      api :PUT, '/ansible_playbooks/sync', N_('Sync Ansible playbooks')
+      param :proxy_id, :identifier, :required => true, :desc => N_('Smart Proxy to sync from')
+      param :playbooks_names, Array, N_('Ansible  playbooks names to be synced')
+      def sync
+        params = @importer.import_playbooks(playbooks_names)
+        if changed?(params)
+          ForemanTasks.async_task(ImportPlaybooksJob::Async::SyncPlaybooks, params[:changed], @ansible_proxy, playbooks_names)
```

Comment:
Since the rabl view uses `@task`, you need to assign the task object into the instance variable.